### PR TITLE
drivers: i3c: cdns: use deterministic timeout for idle

### PR DESCRIPTION
**Pull request by XenuIsWatching**
Previously, the idle bit would be read for X amount of times. This could vary alot depend on the CPU speed. Timeout is now to happen from the cycle time.

### CODE REVIEW

File: drivers/i3c/i3c_cdns.c [hunk 1]
```diff
@@ -371,13 +371,15 @@
...
-#define I3C_MAX_IDLE_WAIT_RETRIES		5000
-#define I3C_PRESCL_REG_SCALE			(4)
-#define I2C_PRESCL_REG_SCALE			(5)
+#define I3C_MAX_DEVS                     11
+#define I3C_MAX_MSGS                     10
+#define I3C_SIR_DEFAULT_DA               0x7F
+#define I3C_MAX_IDLE_CANCEL_WAIT_RETRIES 50
+#define I3C_PRESCL_REG_SCALE             (4)
+#define I2C_PRESCL_REG_SCALE             (5)
+#define I3C_WAIT_FOR_IDLE_STATE_US       20
```
Comment (dcpleung) [inline drivers/i3c/i3c_cdns.c:380]: Could you skip the space changes? These are not related to the idle change.

**Review by dcpleung — DISMISSED** [review]

File: drivers/i3c/i3c_cdns.c [hunk 2]
```diff
@@ -375,9 +375,10 @@
 #define I3C_MAX_MSGS				10
 #define I3C_SIR_DEFAULT_DA			0x7F
 #define I3C_MAX_IDLE_CANCEL_WAIT_RETRIES	50
-#define I3C_MAX_IDLE_WAIT_RETRIES		5000
 #define I3C_PRESCL_REG_SCALE			(4)
 #define I2C_PRESCL_REG_SCALE			(5)
+#define I3C_IDLE_TIMEOUT_CYC                                                                       \
+	(I3C_WAIT_FOR_IDLE_STATE_US * (sys_clock_hw_cycles_per_sec() / USEC_PER_SEC))
```
Comment (dcpleung) [inline drivers/i3c/i3c_cdns.c:381]: Is `I3C_WAIT_FOR_IDLE_STATE_US` specified somewhere? I can't seem to find it via grep.

Comment (XenuIsWatching) [reply drivers/i3c/i3c_cdns.c:382]: yeah... when I fixed the space'ing accidently removed it :(

It's fixed now

**Review by dcpleung — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]
